bib/cmd/bootc-image-builder/main.go

```diff
@@ -315,6 +333,7 @@ func run() error {
 	manifestCmd.Flags().String("config", "", "build config file")
 	manifestCmd.Flags().String("type", "qcow2", "image type to build [qcow2, ami]")
 	manifestCmd.Flags().Bool("tls-verify", true, "require HTTPS and verify certificates when contacting registries")
+	manifestCmd.Flags().String("target-arch", "", "build for the given target architecture")
```

Comment:
shouldn't we specify that you can only enter `x86_64` and `aarch64` ? since that's what's being added to `entrypoint.sh`: https://github.com/osbuild/bootc-image-builder/pull/139/files#diff-6f9d41d046756f0ddc2fcee0626bdb50100d12b88f293734eff742818e03efa2R6

Reply:
Thanks for your review - indeed, the code in this area is a bit loose right now. In theory (if the host has qemu-user{,-static} installed) this will just work magically for all target arches that the container is build for.

However the user experience if the host does not have qemu-user{,-static} is bad hence I added it to the entrypoint.sh but I am having second thoughts about it. The fact that we have to mess with the (non-namespaced, system-wide) binfmt_misc from inside the container is not great. There could also be confusing bugs, because the binfmt_misc registration including the interpreter path is global so if someone has a custome /usr/local/lib/qemu-binfmt-aarch64 or something inside the container this path will not exist and fun errors happen. We also cannot "fix" it from inside the contianer without messing up the host. I think this is a long winded way of saying that we may simply need to error if binfmt_misc is missing and ask the user kindly to install it system-wide. Podman-machine of course can ensure this is avaiable inside the podman machine it creates.

Reply:
Yes I think this is the users or someone outside of the container images responsibility.  If I am attempting to build a different Arch then the default arch without qemu-user-static-ARCH then then podman build will blow up, long before we attempt to convert the image.  You should just pass the arch down and allow the apps to blow up.
Do not attempt to modify the host system, that would be bad behavior and very unexpected.

Reply:
Thank you again for your feedback! I updated the code to not do anything with binfmt_misc from inside the container. For now it will just print a warning that qemu-user is required, we could do something elaborate here by detecting this but that seems like a step(2) feature.